### ISSUE TITLE
How to stop / clean up?

### ISSUE THREAD

**Issue by allefeld** [haugene/docker-transmission-openvpn#1625]
**Before creating this issue I have:**

- [x ] Searched for [similar issues](https://github.com/haugene/docker-transmission-openvpn/issues)
- [ x] [Read the documentation](https://haugene.github.io/docker-transmission-openvpn/) to a reasonable degree

**Describe your request**

The documentation explains how to run the image, but not how to stop it and clean up left-behind data. I figured out stopping with `docker stop`. But even if stopped, the container and associated volumes are left behind. For Docker use, is it recommended to run `docker container prune` and `docker volume prune` regularly, e.g. in the shell script I use to start the image? Or should I use the `--rm` option to `docker run` (not sure whether that also removed volumes)?

**Contextual information**

I have no previous experience with Docker, only came to it because I wanted to use this image. I've read up on it, but Docker is a very complicated piece of software. Since you've explained how to start, I think it would be great if you could explain how to stop & clean up, too.

**Comment by clement-z** [comment]
Well this is really not the job of this project. The docker project itself is quite well documented. See for example [one of the tutorials](https://docs.docker.com/get-started/03_updating_app/) on stopping and removing a container (the whole series is a quick read and will give you an overview of docker).

The command you are looking for is [`rm`](https://docs.docker.com/engine/reference/commandline/rm/), which deletes the specified container. Using `--rm` has the effect to run an equivalent `docker rm` after the container is stopped.

Associated volumes are either:

- "built-in" from the container definition (and live by default at `/var/lib/docker/overlay2/<hash>`);
- a separate entity that you created yourself with `docker volume create my_vol` and bound to the container using `-v my_vol:/data`; the data either lives in `/var/lib/docker/volumes/<hash>` or where you specified it when you created the volume;
- a folder on the host that you shared with the container with e.g. `-v /absolute/path/to/host/data:/data`.

AFAIK, the first one only is deleted after you remove the container. The volume is specifically there to provide persistent storage for docker containers, and shared folders are also a way to achieve that. If you want to get rid of them, you have to delete them (`docker volume rm my_vol` or `rm -rf /absolute/path/to/host/data`)

**Comment by allefeld** [comment]
Yes, there is extensive documentation on Docker, and I read a great deal of it. Sorry if I was unclear, but I wasn't asking to reproduce that. My question is: If I start your image in the way described in the documentation, what would you recommend for stopping and cleanup?

Basically, assuming you use the image yourself on a regular basis, how do you do it? Do you use the `docker run` command only once, and then pause and restart the container, or do you use `docker run` each time and remove the container & voumes each time?

**Comment by pkishino** [comment]
The container is up and running perpetually until a new update is available, then I destroy and recreate.
Please close as this isn’t an issue with the project and more a support question for docker usage

**Comment by clement-z** [comment]
@allefeld As long as the container's `/data` is mounted on a host path, it will survive the docker being removed and recreated.

So the explicit order is: `docker run` → until new update is available → `docker stop` → `docker rm` → `docker pull` → docker run`.

There are options to monitor the docker registry to automatically pull the new image, but this is really out of scope. Another alternative is wrapping all of this in a systemd unit, with something along the lines of:

```
ExecStartPre=-docker rm [...]
ExecStartPre=docker pull [...]
ExecStart=docker run [...]
ExecStop=docker stop [...]
```

**Comment by allefeld** [comment]
Thanks both!

I would recommend to put something along these lines into the documentation.
